jmeter-service/src/main/java/io/eventuate/examples/tram/ordersandcustomers/jmeter/TestPlanController.java

```diff
@@ -0,0 +1,95 @@
+package io.eventuate.examples.tram.ordersandcustomers.jmeter;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.InputStreamResource;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.*;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@RestController
+public class TestPlanController {
+
+  @Value("${jmeter.service.results.folder}")
+  private String resultsFolder;
+
+  @Value("${jmeter.service.testplan.path}")
+  private String testPlanPath;
+
+  @Value("${jmeter.path}")
+  private String jmeterPath;
+
+  @RequestMapping(value = "/testplan", method = RequestMethod.POST)
+  public synchronized String runTestPlan(@RequestParam("loops") int loops,
+                            @RequestParam("threads") int threads,
+                            @RequestParam("rampTime") int rampTime,
+                            @RequestParam("domain") String domain,
+                            @RequestParam("port") int port,
+                            HttpServletRequest request) throws IOException, InterruptedException {
+
+    String testPlanId = UUID.randomUUID().toString();
+
+    configureTestPlan(loops, threads, rampTime, domain, port);
+
+    runTestPlan(testPlanId);
+
+    return String.format("%s/%s/index.html", request.getRequestURL(), testPlanId);
+  }
+
+  @RequestMapping(value = "/testplan/**", method = RequestMethod.GET, produces = MediaType.ALL_VALUE)
+  public @ResponseBody InputStreamResource getFile(HttpServletRequest request) throws IOException {
+      return new InputStreamResource(new FileInputStream(getReportFilePath(request.getRequestURI())));
+  }
+
+  private void runTestPlan(String testPlanId) throws IOException, InterruptedException {
+    File testPlanReportFolder = new File(String.format("%s/%s", resultsFolder, testPlanId));
+    testPlanReportFolder.mkdir();
+
+
+    ProcessBuilder processBuilder = new ProcessBuilder();
+    processBuilder.command(jmeterPath, "-n", "-t", testPlanPath, "-l", getTestResultPath(testPlanId));
+    processBuilder.start().waitFor();
+
+    processBuilder = new ProcessBuilder();
+    processBuilder.command(jmeterPath, "-g", getTestResultPath(testPlanId), "-o", testPlanReportFolder.getAbsolutePath());
+    processBuilder.start().waitFor();
```

Comment:
I don't think it's a good idea to wait for completion.
Perhaps it will take a long time.
This should run in the "background".

Reply:
I will fix